File: 6 – Practice/leet code/014. Longest Common Prefix.cpp
// https://leetcode.com/problems/longest-common-prefix/

class Solution {
public:
    string longestCommonPrefix(vector <string> &strs) {
        sort(strs.begin(), strs.end());

        string l = strs[0];
        string r = strs.back();

        int i = 0;
        const int SIZE = l.size();

        for (; i < SIZE; ++i) {
            if (l[i] != r[i]) {
                break;
            }
        }

        return l.substr(0, i);
    }
};

File: 6 – Practice/leet code/014. Longest Common Prefix.java
// https://leetcode.com/problems/longest-common-prefix/

class Solution {
    public String longestCommonPrefix(String[] strs) {
        Arrays.sort(strs);

        final int SIZE = strs.length;
        int minLen = strs[0].length();

        String l = strs[0];
        String r = strs[SIZE - 1];

        int i = 0;
        for (; i < minLen; ++i) {
            if (l.charAt(i) != r.charAt(i)) {
                break;
            }
        }

        return l.substring(0, i);
    }
}